src/databricks/labs/ucx/framework/logger.py

```diff
@@ -56,6 +62,33 @@ def format(self, record: logging.LogRecord):  # noqa: A003
         return f"{self.GRAY}{ts}{self.RESET} {level} {color_marker}[{name}] {msg}{self.RESET}"
 
 
+@dataclass
+class ObjectFailure:
+    step_name: str
+    object_type: str
+    object_id: str
+    event_time: datetime.now()
+
+
+class FailureReporter:
```

Comment:
I think it belongs more to crawlers.py

Reply:
Yeah, I moved it already into its own file `failures.py`.